spec/lib/puma/appsignal_spec.rb

```diff
@@ -23,6 +23,24 @@ module Puma
       def self.stats
       end
 
+      def self.run
+        # Capture threads running before application is preloaded
+        before = Thread.list.reject { |t| t.thread_variable_get(:fork_safe) }
+
+        # An abbreviated version of what happens in Puma::Cluster#run
+        launcher = MockPumaLauncher.new
+        plugin = Plugin.plugin.new
+        plugin.start(launcher)
+        launcher.events.on_booted.call
+
+        # Capture any new threads running after application is preloaded.
+        # Any threads created during the preloading phase will not be
+        # carried over into the forked workers. Puma warns about these
+        # but the minutely probe thread should only exist in the main process.
+        after = Thread.list.reject { |t| t.thread_variable_get(:fork_safe) }
+        $stdout.puts "! WARNING: Detected #{after.size - before.size} Thread(s) started in app boot" if after.size > before.size
```

Comment:
This line length feels horrible for a trailing `if` but rubocop complains if I do `if ... end`

Reply:
It's okay for now 👍